src/modules/github/contexts/selected-pulls-provider.tsx

```diff
@@ -327,6 +335,51 @@ const selectedPullsReducer = (
       }
     }
 
+    case 'select-all-dependabot': {
+      if (!state.repos) {
+        throw new Error('There is no repos. This should not happen.')
+      }
+
+      const isTargetRepo = (repo: TRepo) =>
+        repo.owner === payload.owner && repo.name === payload.repo && repo.pulls
+      const hasTargetRepo = state.repos.some((repo) => isTargetRepo(repo))
+
+      if (!hasTargetRepo) {
+        return state
+      }
+
+      return {
+        ...state,
+        repos: state.repos.map((repo) => {
+          if (isTargetRepo(repo)) {
+            const isTargetPull = (pull: TPull) =>
+              pull.number === payload.pullNumber
+            const hasTargetPull = repo.pulls!.some((pull) => isTargetPull(pull))
+
+            if (!hasTargetPull) {
+              return repo
+            }
+
+            return {
+              ...repo,
+              pulls: repo.pulls!.map((pull) => {
```

Comment:
How can you know `repo.pulls` are valid? Shouldn't you add exception logic for that?